example/client/main.cpp

```diff
@@ -0,0 +1,225 @@
+        }
+
+        resolver_.async_resolve(
+            host, port, std::bind(&session::on_resolve, this, _1, _2));
+    }
+
+private:
+    void
+    fail(std::string what, boost::system::error_code ec)
+    {
+        if(ec == asio::error::operation_aborted)
+            return;
+
+        std::cerr << what << ": " << ec.message() << "\n";
+    }
+
+    void
+    on_resolve(
+        boost::system::error_code ec,
+        tcp::resolver::results_type results)
+    {
+        if(ec)
+            return fail("async_resolve", ec);
+
+        asio::async_connect(
+            stream_.lowest_layer(),
+            results,
+            std::bind(&session::on_connect, this, _1, _2));
+    }
+
+    void
+    on_connect(
+        boost::system::error_code ec,
+        tcp::resolver::results_type::endpoint_type)
+    {
+        if(ec)
+            return fail("async_connect", ec);
+
+        stream_.async_handshake(
+            ssl::stream_base::client,
+            std::bind(&session::on_handshake, this, _1));
+    }
+
+    void
+    on_handshake(boost::system::error_code ec)
+    {
+        if(ec)
+            return fail("async_handshake", ec);
+
+        sr_.start(req_);
+
+        io::async_write(
+            stream_, sr_, std::bind(&session::on_write, this, _1, _2));
+    }
+
+    void
+    on_write(boost::system::error_code ec, std::size_t /*bytes_transferred*/)
+    {
+        if(ec)
+            return fail("async_write", ec);
+
+        pr_.reset();
+        pr_.start();
+
+        io::async_read_header(
+            stream_, pr_, std::bind(&session::on_read_header, this, _1, _2));
+    }
+
+    void
+    on_read_header(
+        boost::system::error_code ec,
+        std::size_t /*bytes_transferred*/
+    )
+    {
+        if(ec)
+            return fail("async_read_header", ec);
+
+        io::async_read(
+            stream_, pr_, std::bind(&session::on_read_body, this, _1, _2));
+    }
+
+    void
+    on_read_body(
+        boost::system::error_code ec,
+        std::size_t /*bytes_transferred*/
+    )
+    {
+        if(ec)
+            return fail("async_read", ec);
+
+        std::cout << pr_.body() << std::endl;
+
+        stream_.async_shutdown(std::bind(&session::on_shutdown, this, _1));
+    }
+
+    void
+    on_shutdown(boost::system::error_code ec)
+    {
+        if(ec && ec != ssl::error::stream_truncated)
+            return fail("async_shutdown", ec);
+    }
+};
+
+int
+main(int argc, char* argv[])
+{
+    try
+    {
+        // Check command line arguments.
+        if(argc != 2)
+        {
+            std::cerr << "Usage: http-client-async <url>\n"
+                      << "Example:\n"
+                      << "    http-client-async http://www.example.com\n"
+                      << "    http-client-async https://www.example.com/xyz\n";
+            return EXIT_FAILURE;
+        }
+
+        urls::url_view url{ argv[1] };
+        auto port = [&]() -> core::string_view
```

Comment:
I don't think you need this. Asio is perfectly happy if you pass "http" or "https" into the resolver as a port number. It will figure it out from the local configuration. And even if you did need to do this, I believe Boost.URL already has an algorithm for it? @alandefreitas would know.